subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseModel.java

```diff
@@ -95,11 +116,21 @@ public void setClasspath(EclipseClasspath classpath) {
      * For examples see docs for {@link EclipseProject}
      */
     public EclipseJdt getJdt() {
-        return jdt;
+        return jdt.getOrNull();
     }
 
     public void setJdt(EclipseJdt jdt) {
-        this.jdt = jdt;
+        this.jdt.set(jdt);
+    }
+
+    /**
+     * See {@link #setJdt(EclipseJdt)}.
+     *
+     * @since 4.9
+     */
+    @Incubating
+    public void setJdt(Provider<EclipseJdt> jdt) {
```

Comment:
We should just create the EclipseJdt instance eagerly and attach it to the model directly rather than via a provider.